packages/core-mobile/app/screens/nft/send/NftSend.tsx

```diff
@@ -57,15 +57,25 @@ export default function NftSend({
     error,
     isSending,
     isValid,
-    isValidating
+    setCanValidate
   } = useSendContext()
   const { activeNetwork } = useNetworks()
   const activeAccount = useSelector(selectActiveAccount)
   const fromAddress = activeAccount?.addressC ?? ''
   const tokens = useSelector(selectTokensWithBalance)
-  const nativeToken = tokens.find(
-    t => t.type === TokenType.NATIVE
-  ) as NetworkTokenWithBalance
+  const [nativeToken, setNativeToken] = useState<NetworkTokenWithBalance>(
+    tokens.find(t => t.type === TokenType.NATIVE) as NetworkTokenWithBalance
+  )
+  // in production, balance fetching occurs every two seconds, so update native token only when balance changes to avoid unnecessary re-renders
+  useEffect(() => {
+    const updatedNativeToken = tokens.find(
+      t => t.type === TokenType.NATIVE
+    ) as NetworkTokenWithBalance
+    if (!nativeToken || updatedNativeToken?.balance !== nativeToken.balance) {
+      setNativeToken(updatedNativeToken)
+    }
+  }, [tokens, nativeToken])
```

Comment:
let's create a hook for this to reuse since we have to do it in multiple places. useNativeToken() or something.

Reply:
added `useNativeTokenWithBalance` hook.